src/lib.rs

```diff
@@ -1,28 +1,123 @@
 use std::fmt::Write;
 
-#[derive(PartialEq, Eq)]
-pub struct Roff {
+/// Title line for a manpage.
+pub struct Title {
     title: String,
-    section: i8,
+    section: ManSection,
+    date: Option<String>,
+    source: Option<String>,
+    manual: Option<String>,
+}
+
+impl Title {
+    pub fn new(title: &str, section: ManSection) -> Self {
+        Title {
+            title: title.into(),
+            section,
+            date: None,
+            source: None,
+            manual: None,
+        }
+    }
+}
+
+impl Troffable for Title {
+    fn render(&self) -> String {
+        let manual = self.manual.as_deref().unwrap_or_default();
+        let date = self.date.as_deref().unwrap_or_default();
+        let source = self.source.as_deref().unwrap_or_default();
+
+        format!(
+            r#".TH "{}" "{}" "{}" "{}" "{}""#,
+            self.title.to_uppercase(),
+            self.section.value(),
+            date,
+            source,
+            manual
+        )
+    }
+}
+
+/// Manpage sections.
+///
+/// The most common is [`ManSection::Executable`], and is the recommended default.
+#[derive(Clone, Copy)]
+pub enum ManSection {
+    /// Executable programs or shell commands
+    Executable,
+    /// System calls (functions provided by the kernel)
+    SystemCalls,
+    /// Library calls (functions within program libraries)
+    LibraryCalls,
+    /// Special files (usually found in /dev)
+    SpecialFiles,
+    /// File formats and conventions, e.g. /etc/passwd
+    FileFormats,
+    /// Games
+    Games,
+    /// Miscellaneous (including macro packages and conventions), e.g. man(7), groff(7)
+    Miscellaneous,
+    /// System administration commands (usually only for root)
+    SystemAdministrationCommands,
+    /// Kernel routines [Non standard]
+    KernelRoutines,
+}
+
+impl ManSection {
+    pub fn value(&self) -> i8 {
+        match self {
+            ManSection::Executable => 1,
+            ManSection::SystemCalls => 2,
+            ManSection::LibraryCalls => 3,
+            ManSection::SpecialFiles => 4,
+            ManSection::FileFormats => 5,
+            ManSection::Games => 6,
+            ManSection::Miscellaneous => 7,
+            ManSection::SystemAdministrationCommands => 8,
+            ManSection::KernelRoutines => 9,
+        }
+    }
+}
+
+pub struct Roff {
```

Comment:
If `Roff` is the entry point for this crate. please keep it first in the file so its easier to jump in and see what this crate does

Reply:
Will fix!